src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -341,11 +353,41 @@
     (->> (map format-task info-from-jruby)
          (middleware-utils/json-response 200))))
 
+(schema/defn ^:always-validate
+  describe-task :- TaskDetails
+  [task-data :- TaskData
+   env-name :- schema/Str
+   module-name :- schema/Str]
+  (let [?metadata (try (some-> task-data :metadata_file slurp json/decode)
+                   (catch Exception e
+                     nil))
+        describe-file (fn [file]
```

Comment:
I know in the classifier it was a pattern to make anonymous functions for fns that only get used in that one function, but it's not my own perference, as it makes functions longer. What would you think about breaking this out?

Reply:
something like `generate-task-metadata` or something? That would also be nicely unit testable.

Reply:
I'll split it out